Adding content and cleaning up some of the build process

jimschubert opened the pull request:
Hey TJ,

I've been working on adding some content to Mastering Node and updating it to v0.4.x.  It's pretty well-organized and stubbed out, and I think when the empty sections are completed it will be a necessity in the toolset of node.js beginners.  In fact, I found the book while looking for resources because I wanted to start learning node.

I've cleaned up the build process and structure a little.  For instance, I removed the ronn dependency in favor of using markup-js and the doctool.js that is available in the current joyent/node/tools/doctools directory.  This is really clean and I plan to modify the file to include more options.  I also mimicked the node.js markup structure (using @include) to move the structure of the document out of the make file into chapters/all.md.

My next task, aside from adding more content, is to look for (or write) an html to pdf conversion module written in node.js to remove the htmldoc dependency.  I think that having the book about node.js being compiled entirely by node scripts is completely meta.  When i was in programming classes in college, one of my instructors gave us the answers to a test as code that we had to compile and run for the answers.  I think it fires up interest, more than anything.

The main thing I did to the structure was to move the book.\* files to a /doc directory.  This would change the gh-pages references, which I didn't touch.

Thanks,

Jim

jimschubert committed:
Modified module example to use require.extensions, because require.registerExtension() was removed in node 0.3.0

jimschubert committed:
Updated modules chapter to reflect node 0.3.0 remove of require.registerExtension

jimschubert committed:
Updated installation instructions in markdown file

jimschubert committed:
Updated modules.md by explaining how to example pseudo-globals in node console

jimschubert committed:
Updated section on globals: console

jimschubert committed:
Updated some grammar and functionality for globals: process

jimschubert committed:
Updated events and buffers, mostly grammar and formatting. These sections still need to be expanded a little more.

jimschubert committed:
Regenerated book files and updated makefile slightly

jimschubert committed:
Updated Readme.md to show Jim Schubert as an author and introduce repl (node console)

jimschubert committed:
Cleaning up the project and make process

jimschubert committed:
Regenerated book

jimschubert committed:
Removed generated html files which don't need to be tracked by git

jimschubert committed:
Updated .gitignore to ignore the html files generated from markdown

jimschubert committed:
Updated readme to support multiple authors ;)

jimschubert committed:
Updated book.html generation to reflect the removal of ronn dependency.

jimschubert committed:
Added an events example from the nodejs documentation which provides a little more insight on the nature of events

jimschubert committed:
Added to the intro and cleaned up the make process a little

jimschubert committed:
removed redundant files and test files. Whoops

jimschubert committed:
Removed toc.md, which was replaced by all.md and auto-generated tocs.  Updated styles

jimschubert committed:
Added an initial Hello World example like we're basically obligated to do.  Added a simple tcp example

tj commented:
cool man nice stuff, might be a little but until I can review, but thanks! appreciated :)

jimschubert committed:
Added meta to html to specify encoding, description, keywords, author(s)

jimschubert committed:
Added example standards explanation to intro.md

jimschubert committed:
modified .gitignore to ignore files and directories added by ndistro

jimschubert committed:
Updated Makefile to clean/build property for each book. Removed improper style.css comment

jimschubert committed:
Updated Makefile to clean/build property for each book. Removed improper style.css comment
Also, updated installation instructions to remove nDistro (I couldn't get it work properly with the current node release).  Now, installation instructions are based on installing directly from source and using npm to install dependencies throughout the book.

jimschubert committed:
Merge branch 'master' of github.com:jimschubert/masteringnode

jimschubert committed:
Updated intro with more standardized formatting, removed nDistro link from intro, and updated modules to reflect the standardized formatting

jimschubert committed:
Updated intro with more standardized formatting, removed nDistro link from intro, and updated modules to reflect the standardized formatting

jimschubert committed:
globals code conventions

jimschubert committed:
Updated events with standard code block conventions

jimschubert committed:
Some minor grammar edits

jimschubert committed:
Code block standards

jimschubert committed:
fs code block standards

jimschubert committed:
Began on testing chapter and some minor modifications to other chapters

jimschubert committed:
Merge branch 'master' of github.com:jimschubert/masteringnode

Conflicts:
	docs/assets/style.css

jimschubert committed:
Working on http and testing

jimschubert committed:
Started on TCP

jimschubert committed:
Merge branch 'master' of github.com:jimschubert/masteringnode

jimschubert committed:
Added a start to the nodeunit section

jimschubert committed:
Regenerated book

jimschubert committed:
Updated http example to allow for shutting down the server and writing output to the correct directory

jimschubert committed:
Added an example for http server

jimschubert committed:
Merge branch 'master' of github.com:jimschubert/masteringnode

jimschubert committed:
Finished an explanation of the http server example

jimschubert committed:
Updated the mobi output

jimschubert committed:
Added a slightly cleaner server example which emits events

jimschubert committed:
Regenerated book

jimschubert committed:
Updated http and section in globals related to constants

jimschubert committed:
Fixed a buffer example that was missing the buffer creation

jimschubert committed:
Added the addon example from node docs

jimschubert committed:
Added new function example and updated chapter

jimschubert committed:
Regenerated book

jimschubert committed:
Modified some typos and grammar

jimschubert committed:
Regenerated book

jimschubert committed:
Updated Readme.md from visionmedia/masteringnode changes

jimschubert committed:
Regenerated book

jimschubert committed:
Added debugging section and regenerated book

jimschubert committed:
Changed markdown.js to submodule and updated doctool to properly load markdown dep

jimschubert committed:
Regenerated book

jimschubert committed:
Added section on deploying to Heroku, vim configurations, and regenerated books

jimschubert committed:
Added uuid generator node addon example and corresponding update to FunctionTemplate section in addons.md

Regenerated book

jimschubert committed:
crlf

jimschubert committed:
Fix CRLF

jimschubert committed:
Added gh-pages to /site

jimschubert committed:
Trying out some syntax highlighter changes to prettyprint

jimschubert committed:
Regenerated book

jimschubert committed:
Changed background on site and updated fork me link

jimschubert committed:
Added 'make gh-pages' to easily copy /site to gh-pages branch

jimschubert committed:
Merge remote-tracking branch 'origin/master'

* origin/master:
  Added 'make gh-pages' to easily copy /site to gh-pages branch
  Changed background on site and updated fork me link
  Regenerated book
  Trying out some syntax highlighter changes to prettyprint
  Added gh-pages to /site
  Fix CRLF

jimschubert committed:
Updating tcp examples

jimschubert committed:
Updating make gh-pages to allow reading online. Removed syntax highlighter and docs/assets*

jimschubert committed:
Regenerated book
